src/AuthContext.tsx

```diff
@@ -93,21 +93,33 @@ export const AuthProvider = (props: AuthProviderProps) => {
 
     // Create a client and register an observer that triggers when the user logs in or out
     const client = useMemo(() => {
-        const client = createClient({ authUrl: props.authUrl, enableBackgroundTokenRefresh: true })
-        client.addAccessTokenChangeObserver(() => setAccessTokenChangeCounter((x) => x + 1))
-        return client
+        return createClient({ authUrl: props.authUrl, enableBackgroundTokenRefresh: true })
     }, [props.authUrl])
 
-    // On unmount, destroy the client
     useEffect(() => {
-        if (props.getActiveOrgFn) {
-            console.warn("The `getActiveOrgFn` prop is deprecated. Please use `useActiveOrgV2` instead.")
+        const observer = () => {
+            setAccessTokenChangeCounter((x) => x + 1)
+        }
+        client.addAccessTokenChangeObserver(observer)
+        return () => {
+            client.removeAccessTokenChangeObserver(observer)
         }
+    }, [client])
+
+    // On unmount, destroy the client
+    useEffect(() => {
         return () => {
             client.destroy()
         }
     }, [])
 
+    // Deprecation warning
+    useEffect(() => {
+        if (props.getActiveOrgFn) {
+            console.warn("The `getActiveOrgFn` prop is deprecated.")
+        }
+    }, [])
```

Comment:
Since you only want to do this once on the initial rendering of the context/first value of the `getActiveOrgFn`, I think you can put this in the same `useEffect` as the one where `client.destroy` is called

Reply:
Oh yeah, I actually moved it out because I wanted to think of them as conceptually different.